libethdrivers/src/plat/imx6/imx6.c

```diff
@@ -393,10 +395,17 @@ int ethif_imx6_init(struct eth_driver *eth_driver, ps_io_ops_t io_ops, void *con
     }
     eth_data->enet = enet;
 
-    /* Fetch and set the MAC address */
-    if (ocotp == NULL || ocotp_get_mac(ocotp, mac)) {
-        memcpy(mac, DEFAULT_MAC, 6);
+    if (plat_config->prom_mode) {
```

Comment:
This is a bit of a breaking change, given that we have code that passes in `NULL` as the `config` ([as specified in the function description](https://github.com/seL4/libethdrivers/blob/master/include/ethdrivers/imx6.h#L25)). Perhaps the config should be made an optional parameter?

Reply:
Do you have an example using the i.MX6 driver anywhere? We only saw x86 examples. We could work together to figure out ways to improve this.

Reply:
Okay so the example I'm thinking of is from our [Advanced Operating Systems course](http://www.cse.unsw.edu.au/~cs9242/current/), but it is pinned to an old version of the code anyway so doesn't matter. I would be happy with updating the comment in the header and then having an `assert(config)`

Reply:
Alright. We will update the header command and add in the assert value.